chain/txValidator.go

```diff
@@ -488,15 +487,13 @@ func (bvs *BlockValidationState) VerifyTransactionWithBlock(txn *transaction.Tra
 
 	switch txn.UnsignedTx.Payload.Type {
 	case pb.COINBASE_TYPE:
-		if header != nil {
-			coinbase := payload.(*pb.Coinbase)
-			donationAmount, err := DefaultLedger.Store.GetDonation()
-			if err != nil {
-				return err
-			}
-			if Fixed64(coinbase.Amount) != GetRewardByHeight(header.UnsignedHeader.Height)+donationAmount {
-				return errors.New("Mining reward incorrectly.")
-			}
+		coinbase := payload.(*pb.Coinbase)
```

Comment:
Why not skipping the check when height is 0?

Reply:
height is only 0 when called from txpool and txpool don't have coinbase txs

Reply:
I think we should further make sure that by adding a switch condition in pool.processTx that returns error when type is coinbase